docs/content/concepts/code-locations.mdx

```diff
@@ -0,0 +1,147 @@
+---
+title: "Code Locations | Dagster Docs"
+description: ""
+---
+
+# Code Locations
+
+A code location is set of dagster definitions loadable and accessible by Dagster's tools, such as the CLI, Dagit, and Dagster Cloud. A code location comprises:
+
+- A reference to a python module or package that has an instance of `Definitions` at the "defs" variable. We will refer to that as the "entry point".
+- A Python environment that can successfully load the entry point.
+
+Definitions within a code location have a common namespace and must have unique names. This allows them to be grouped and organized by code location in tools.
+
+Before the introduction of the <PyObject object="Definitions" /> API, definitions were grouped into repositories, and there could be many repostories in a particular code location. See the [guide](/concepts/repositories-workspaces/repositories) on Repositories for this previous API and mental model.
```

Comment:
We can wrap this in a note so it won't get in the way for users who don't care what a repository was

```suggestion
<Note>
Before the introduction of the <PyObject object="Definitions" /> API, definitions were grouped into repositories, and there could be many repostories in a particular code location. See the <a href="/concepts/repositories-workspaces/repositories">guide</a> on Repositories for this previous API and mental model.
</Note>
```

Reply:
cool. i'll make this change myself as committing suggestions doesn't play nice with spr